### PR TITLE
Don't split escaped function labels

### PR DESCRIPTION
**Pull request by yaelgen**
A fix for: [ML-3224](https://jira.iguazeng.com/browse/ML-3224)

### CODE REVIEW

File: nuclio/config.py [hunk 1]
```diff
@@ -74,8 +74,26 @@ def get_in(obj, keys):
     return obj
 
 
+def split_by_dots_with_escaping(key: str):
+    """
+    splits the key by dots, taking escaping into account so that the key can contain dots
```
Comment (AlonMaor14) [inline nuclio/config.py:79]:
```suggestion
    splits the key by dots, taking escaping into account so that an escaped key can contain dots
```

File: nuclio/config.py [hunk 2]
```diff
@@ -74,8 +74,26 @@ def get_in(obj, keys):
     return obj
 
 
+def split_by_dots_with_escaping(key: str):
```
Comment (AlonMaor14) [inline nuclio/config.py:77]: can be private
```suggestion
def _split_by_dots_with_escaping(key: str):
```

File: tests/test_utils.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ def test_update_in():
     config.update_in(obj, 'a.b.c', 3)
     assert obj['a']['b']['c'] == 3
 
-    config.update_in(obj, 'a.b.d', 3, append=True)
-    assert obj['a']['b']['d'] == [3]
-    config.update_in(obj, 'a.b.d', 4, append=True)
-    assert obj['a']['b']['d'] == [3, 4]
```
Comment (AlonMaor14) [inline tests/test_utils.py:28]: why delete this? seems like it should still be supported

**Review by AlonMaor14 — CHANGES_REQUESTED** [review]
Nice! I like the logic.
Please add some tests for edge cases like empty, key without dots, etc.

File: nuclio/config.py [hunk 4]
```diff
@@ -74,8 +74,26 @@ def get_in(obj, keys):
...
+    """
+    splits the key by dots, taking escaping into account so that the key can contain dots
+    """
+    parts = []
+    current_key, escape = "", False
+    for char in key:
+        if char == "." and not escape:
+            parts.append(current_key)
+            current_key = ""
+        elif char == "\\":
```
Comment (AlonMaor14) [inline nuclio/config.py:87]: what if char is `\"` ? will it catch it? and if so will it filter the `"`?
what is the expected use case here? is the escaping done by the user?

Comment (yaelgen) [reply nuclio/config.py:87]: It will take it as a regular char in the key, I added a test with a case containing `\"`, the result is:
image

File: tests/test_utils.py [hunk 5]
```diff
@@ -26,3 +28,31 @@ def test_update_in():
...
+            ["spec", "handler"],
+            [1, 2, 3],
+        ),
+        (["metadata", "test", "labels", "test.data"], 1),
+        (["metadata", "ֿ״test", "lables\"", "test.data"], 1),
+        (["metadata.test", "test.test", "labels", "test.data"], True),
+        (["metadata", "test.middle.com", "labels", "test.data"], "data"),
+    ],
+)
+def test_update_in_for_key_with_dots(keys, val):
```
Comment (AlonMaor14) [inline tests/test_utils.py:50]:
```suggestion
def test_update_in_with_dotted_keys(keys, val):
```

**Review by AlonMaor14 — APPROVED** [review]
LGTM! minor naming suggestion
